src/frontend/src/components/ActionDropdown/ActionDropdown.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
+import { Button, Form, Dropdown } from 'react-bootstrap';
+
+export default function ActionDropdown({
+  title,
+  formFields,
+  handleSubmit,
+  buttonText,
+}) {
+  const [formData, setFormData] = useState(
+    formFields.reduce((acc, field) => {
+      acc[field.name] = field.defaultValue || '';
+      return acc;
+    }, {}),
+  );
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({ ...prevData, [name]: value }));
+  };
+
+  const handleFormSubmit = async () => {
+    await handleSubmit(formData);
+  };
+
+  return (
+    <Dropdown>
```

Comment:
Instead of a dropdown would it be possible for it to be a toggle that pushes the other buttons down? its a little wired how the dropdown hides the other buttons

Reply:
Yes 100%, this is now a collapsable form instead of a dropdown